src/domain/dataverse/domain.ts

```diff
@@ -179,7 +198,24 @@ export const storeFactory = (
                 )
               )
             )
-          )
+          ),
+        setByPropertyFilter: (newFilter: ByPropertyFilterInput) => () =>
+          set(state => ({
+            data: {
+              ...state.data,
+              dataverse: [],
+              filters: {
+                ...state.data.filters,
+                byProperty: pipe(
+                  !!newFilter.value.length,
+                  B.match(
+                    () => null,
+                    () => newFilter
```

Comment:
You can also make it more readable by avoiding the double bang operator:
```suggestion
                  newFilter.value,
                  S.isEmpty,
                  B.match(
                    () => newFilter,
                    () => null
```

Reply:
Yes we can also make it this way, for me it was shorter and easy to understand with the `!!`but I can make the change :)